pkg/k8s/serviceAccountHelper.go

```diff
@@ -59,27 +60,37 @@ func (h *serviceAccountHelper) GetServiceAccountSecretNameRepeat(ctx context.Con
 	}(time.Now())
 
 	for {
-		result := h.GetServiceAccountSecretName(ctx)
+		result, err := h.GetServiceAccountSecretName(ctx)
+		if err != nil {
+			return "", err
+		}
 		if result != "" {
-			return result
+			return result, nil
 		}
 		retryCount++
 		time.Sleep(retryInterval)
-		h.Reload(ctx)
+		err = h.Reload(ctx)
+		if err != nil {
+			return "", err
+		}
 	}
 }
 
 // GetServiceAccountSecretName retrieves the name of the service account
 // token secret.
-func (h *serviceAccountHelper) GetServiceAccountSecretName(ctx context.Context) string {
+func (h *serviceAccountHelper) GetServiceAccountSecretName(ctx context.Context) (string, error) {
+	client := h.factory.CoreV1().Secrets(h.cache.GetNamespace())
 	for _, secretRef := range h.cache.Secrets {
-		client := h.factory.CoreV1().Secrets(h.cache.GetNamespace())
 		secret, err := client.Get(ctx, secretRef.Name, metav1.GetOptions{})
-		if err == nil &&
-			secret != nil &&
-			secret.Type == v1.SecretTypeServiceAccountToken {
-			return secret.Name
+		if err != nil {
+			if errorsk8s.IsNotFound(err) {
+				continue
+			}
+			return "", err
```

Comment:
The error could be recoverable but this way no retry is executed.

Reply:
If we want retries on recoverable API call errors, we should have a generic solution that is used anywhere. So far we don't have this and all errors go up to the controller, which aborts the current reconciliation and puts the pipeline run back into the work queue for a later retry.

The purpose _here_ is to synchronize with the K8s control plane which provides a service account secret asynchronously. Not more.